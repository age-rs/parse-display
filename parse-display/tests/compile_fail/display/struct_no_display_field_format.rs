use parse_display::Display;

struct NoDisplay;

#[derive(Display)]
#[display("{x}")]
struct NoDisplayFieldInFormat {
    x: NoDisplay,
}

fn main() {}
