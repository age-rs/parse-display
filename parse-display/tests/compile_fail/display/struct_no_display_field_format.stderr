error[E0277]: `NoDisplay` doesn't implement `std::fmt::Display`
 --> tests/compile_fail/display/struct_no_display_field_format.rs:6:11
  |
5 | #[derive(Display)]
  |          ------- in this derive macro expansion
6 | #[display("{x}")]
  |           ^^^^^ `NoDisplay` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `NoDisplay`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: this error originates in the macro `$crate::format_args` which comes from the expansion of the derive macro `Display` (in Nightly builds, run with -Z macro-backtrace for more info)
